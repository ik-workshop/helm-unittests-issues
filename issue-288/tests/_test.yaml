---
templates:
  - "pod.yaml"
tests:
  - it: should selects the correct kind and match snapshot
    documentSelector:
      path: kind
      value: Pod
    asserts:
      - matchSnapshot: {}
  - it: should validate path exists
    asserts:
      - exists:
          path: spec.affinity.nodeAffinity
