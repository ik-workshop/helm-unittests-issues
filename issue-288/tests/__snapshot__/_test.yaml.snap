should selects the correct kind and match snapshot:
  1: |
    apiVersion: v1
    kind: Pod
    metadata:
      name: with-node-affinity
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: topology.kubernetes.io/zone
                    operator: In
                    values:
                      - antarctica-east1
                      - antarctica-west1
            preferredDuringSchedulingIgnoredDuringExecution:
              - preference:
                  matchExpressions:
                    - key: another-node-label-key
                      operator: In
                      values:
                        - another-node-label-value
                weight: 1
